components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/federatedIdpInitLogout/factory/LogoutResponseFactory.java

```diff
@@ -0,0 +1,111 @@
+ * the authentication framework.
+ */
+public class LogoutResponseFactory extends HttpIdentityResponseFactory {
+
+    @Override
+    public boolean canHandle(IdentityResponse identityResponse) {
+
+        return (identityResponse instanceof LogoutResponse);
+
+    }
+
+    @Override
+    public boolean canHandle(FrameworkException exception) {
+
+        return (exception instanceof LogoutException);
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder create(IdentityResponse identityResponse) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder httpIdentityResponseBuilder = new HttpIdentityResponse
+                .HttpIdentityResponseBuilder();
+        create(httpIdentityResponseBuilder, identityResponse);
+        return httpIdentityResponseBuilder;
+    }
+
+    @Override
+    public void create(HttpIdentityResponse.HttpIdentityResponseBuilder builder, IdentityResponse identityResponse) {
+
+        builder.setStatusCode(((LogoutResponse) identityResponse).getStatusCode());
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_CACHE_CONTROL,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_CACHE_CONTROL_NO_STORE);
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_PRAGMA,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_PRAGMA_NO_CACHE);
+        builder.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN);
+        builder.setBody(((LogoutResponse) identityResponse).getMessage());
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder handleException(FrameworkException frameworkException) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder;
+        if (frameworkException instanceof LogoutClientException) {
+            builder = buildResponse(frameworkException.getMessage(),
+                    HttpServletResponse.SC_BAD_REQUEST);
+        } else {
+            builder = buildResponse(OIDCErrorConstants.ErrorMessages.LOGOUT_SERVER_EXCEPTION.getMessage(),
+                    HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+
+        return builder;
+    }
+
+    /**
+     * Build Identity response based on the error message and code.
+     *
+     * @param errorMessage
+     * @param errorCode
+     * @return
+     */
+    private HttpIdentityResponse.HttpIdentityResponseBuilder buildResponse(String errorMessage, int errorCode) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder =
+                new HttpIdentityResponse.HttpIdentityResponseBuilder();
+        builder.setBody(errorMessage);
+        builder.setStatusCode(errorCode);
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_CACHE_CONTROL,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_CACHE_CONTROL_NO_STORE);
+        builder.addHeader(OAuthConstants.HTTP_RESP_HEADER_PRAGMA,
+                OAuthConstants.HTTP_RESP_HEADER_VAL_PRAGMA_NO_CACHE);
+        builder.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN);
```

Comment:
Is this the correct content type?

Reply:
There's no mention of the content type for the logout response in the OIDC specification.